src/cli/index.js

```diff
@@ -22,6 +22,11 @@ function run(args) {
       process.exit(1);
     }
 
+    if (context.argv["only-changed"] && !context.argv["write"]) {
+      context.logger.error("Cannot use --only-changed without --write.");
+      process.exit(1);
+    }
```

Comment:
Maybe we should enable `cache` by default and better add option `--no-cache` (better DX)

Reply:
I thought it might be best to avoid changing existing behavior silently, especially since this flag writes a file that needs to gitignored. I'll defer to the maintainers, y'all know best.

Reply:
/cc @prettier/core

Reply:
Maybe we can do this for `v2`

Reply:
Yeah, probably better to do it for v2

Reply:
But what is problem implement `--no-cache` by default?